fairscale/optim/oss.py

```diff
@@ -145,14 +145,20 @@ def partition_parameters(self) -> List[List[dict]]:
         return self._partition_parameters
 
     @property
-    def local_params(self) -> Iterable[torch.Tensor]:
+    def local_params(self) -> List[torch.Tensor]:
+        """ Iterable which goes through the parameters that this rank owns
+        """
         if self._local_params is None:
-            self._local_params = chain(
```

Comment:
the catch in https://github.com/facebookresearch/fairscale/pull/363 was that I had the wrong mental model for this in Python, I thought that self._local_params was just the top of a linked list, which was never reassigned -> always pointing to the right place. That's actually completely wrong, all the elements in the chain are consumed so this only worked for the first step through